server/server_test.go

```
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"testing"
+)
+
+type TestCase struct {
+	Name           string
+	Phone          string
+	ExpectedStatus int
+	ExpectedError  string
+}
+
+func TestSetData(t *testing.T) {
+	testCases := []TestCase{
+		{
+			Name:           "Igor",
+			Phone:          "1112245",
+			ExpectedStatus: http.StatusOK,
+		},
+		{
+			Name:           "Igor",
+			Phone:          "",
+			ExpectedStatus: http.StatusBadRequest,
+			ExpectedError:  "Params name or phone undefined",
+		},
+		{
+			Name:           "",
+			Phone:          "11548784",
+			ExpectedStatus: http.StatusBadRequest,
+			ExpectedError:  "Params name or phone undefined",
+		},
+	}
+
+	for ind, currTest := range testCases {
+		req := &http.Request{
+			Method: http.MethodPost,
+		}
+
+		req.URL, _ = url.Parse("http://127.0.0.1:8098/set")
+		q := req.URL.Query()
+
+		q.Set("name", currTest.Name)
+		q.Set("phone", currTest.Phone)
+
+		req.URL.RawQuery = q.Encode()
+		resp, err := http.DefaultClient.Do(req)
+
+		if err != nil {
+			t.Errorf("Error connecting to server")
+		}
+		defer resp.Body.Close()
+
+		respBody, _ := ioutil.ReadAll(resp.Body)
+		var error string
+		json.Unmarshal(respBody, &error)
+
+		if currTest.ExpectedStatus != resp.StatusCode && currTest.ExpectedError != error {
+			t.Errorf("[%v] Unexpected status got: %v. Error: %v\n", ind, resp.StatusCode, error)
+		}
+	}
+}
+
+func TestGetData(t *testing.T) {
+	req := &http.Request{
+		Method: http.MethodPost,
+	}
+
+	req.URL, _ = url.Parse("http://127.0.0.1:8098/get")
```

I assume here is error ignored
Please never ignore errors